src/Ocelot/Configuration/Creator/RoutesCreator.cs

```diff
@@ -76,6 +78,13 @@ public List<Route> Create(FileConfiguration fileConfiguration)
                 .ToList();
         }
 
+        public int CreateTimeout(FileRoute fileRoute, FileGlobalConfiguration globalConfiguration)
+        {
+            return fileRoute.Timeout > 0 
+                ? fileRoute.Timeout.Value 
+                : (globalConfiguration.TimeoutSeconds ?? DefaultRequestTimeoutSeconds) * 1000;

```

Comment:
Using ms everywhere is probably a better choice.

Reply:
@ggnaegi Gui, having `ms` dimension for timeout logic will 
- confuse developers 
- introduce breaking change for current QoS 

2nd argument, the [Retry-After](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) header spec has seconds as dimension, not milliseconds!
Also [Keep-Alive](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Keep-Alive) header has seconds as a dimension.

So, I don't recommend to use milliseconds for timeouts.

P.S. But for QoS we use millisecs: [QoSOptions](https://ocelot.readthedocs.io/en/latest/search.html?q=QoSOptions). It's OK because QoS should have millisecs precision for settings.